### DISCUSSION TITLE
OpenLane vs OpenROAD

### DISCUSSION THREAD

**Discussion by Sinbiosis** [The-OpenROAD-Project/OpenLane#900] [Q&A] [answered]
Could someone explain to me what the differences are between OpenLane and OpenROAD? Especially since I can get OpenROAD to work using the OpenROAD-flow-scripts repo and OpenLane repo works for a co-worker of mine. 

I've found historical articles like [this (1)](https://www.design-reuse.com/news/50383/efabless-openroad-openlane.html) that say OpenLane is Efabless' flow and I read in the OpenROAD READMEs that OpenROAD came from DARPA. However, the OpenROAD group "has adopted Efabless’ OpenLane as the flow of record for the OpenROAD program" (1). If so why the need to update the OpenROAD-flow-scripts repo? 

Also, I found [this paper on IEEE from 2020](https://ieeexplore.ieee.org/document/9256623) that has, in figure 2, OpenROAD as a subsection of the full OpenLane RTL-to-GDS flow. I feel like that figure 2 is outdated since it seems OpenROAD now uses yosys/abc and other programs for steps beyond what figure 2 shows OpenROAD accomplishing.

**Comment by Sinbiosis** [comment]
I found [a comment here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/discussions/284) by @maliberty about the current relationship between OpenROAD (OR), OpenLane (OL), SiliconCompiler (SC), and Chipyard. For the purpose of future proofing, would it be best to go with OL as the defacto flow? Assuming OR and OL combine efforts into the OL repo.

**Reply by abettati** [reply]
I have the same question regarding future proofing.

**Reply by maliberty** [reply]
The plan to merge ORFS with OL has never come to fruition.  At this point the OR team is looking more at [silicon compiler](https://github.com/siliconcompiler/siliconcompiler).

**Reply by abettati** [reply]
Thanks Matt for the update. My understanding is that SC is python based, so will it handle the translation to tcl at the OR interface?

**Reply by maliberty** [reply]
Yes

**Reply by magiwanders** [reply]
What do you mean exactly by  "OR team looking at Silicon Compiler"? SC docs mention tapeout output, which means SC should be an alternative to the whole ORFS/OL workflow (RTL to GDSII). Does this mean eventual abandonment of OR or did I understand something wrong?

**Reply by abettati** [reply]
@magiwanders at the moment OR is liisted among the supported technologies in the SC readme.

**Reply by magiwanders** [reply]
Oh I missed it. Thank you, the relationship between the two is now clear :)

**Reply by maliberty** [reply]
SC *could* eventually replace ORFS.  OL is from a separate team so I imagine it would continue regardless.  OR is common to OL, ORFS, and SC.

**Comment by Sinbiosis** [comment]
It seems, based on [this video here](https://www.youtube.com/watch?v=Vhyv0eq_mLU), that OpenLane was originally meant to be OpenROAD, just optimized for the skywater 130nm PDK. At least going by Mohamed Shalan, the OpenLane original developer. So, the sum and gist of OpenLane vs OpenROAD is, "They are the same, just OpenLane is specific to skywater 130nm PDK, but the OpenROAD team is combining OpenROAD and OpenLane together."

**Reply by Sinbiosis** [reply]
So, for future proofing, OL is the better choice.

**Comment by maliberty** [comment]
Just to clarify OpenROAD is separate from OpenROAD-flow-scripts.  The former is the application and the later is scripts to use the application to process a design (along with yosys & klayout).  It is ORFS that is going to merge into OL eventually.  OR itself will remain.

**Comment by donn** [comment, marked as answer]
So, to update this answer:

OpenROAD is one of the tools used by the OpenLane flow. The former is developed primarily by the OpenROAD project, the latter is developed primarily by Efabless Corporation.

There was an attempt a while back to ensure both Efabless and OpenROAD converge on the same flow (the OpenROAD Project uses OpenROAD Flow Scripts), but we've ultimately decided against it.

**Comment by Wiki1998-dev** [comment]
Hi, Above conversation was really informative! I'm currently trying to transition the X-Heep project (which uses OpenROAD flow scripts) to OpenLane, but I'm running into multiple issues and I'm not quite sure how to proceed. Does anyone have any advice or experience with this?
@donn @magiwanders @maliberty @abettati

**Reply by maliberty** [reply]
I'm curious why you are migrating.  What is the motivation?  

My experience with trying to migrate a long time ago was that it was very hard to add news PDKs to OL.  The OL flow tends to lag ORFS and so many new features were missing. Perhaps OL2 has improved the situation.

**Reply by donn** [reply]
I'm also interested- but also, we've been staking out new additions to OpenROAD and ORFS-- what major new features are missing?

**Reply by maliberty** [reply]
I don't track OL enough to know the gaps.  In general we try to make new features on by default so that other flows pick them up however some require flow changes.  Do you use rtl_macro_placer?  If so, are you keeping any hierarchy in yosys?  Do you support pad generation or only Caravel?  Post-grt timing repair with incremental global routing?

**Reply by donn** [reply]
Nope- we'll start tracking the first one and last one. Thanks for the heads up!
